ANR crash when SQLite connection closed by DBManager.setTalkProgress()

Noticed an App Not Responding crash during these conditions:
1. Actively playing a talk, and
2. Scrolling through a long Talks or Teachers list

Then I tried to reproduce by scrolling rapidly as talk is playing, now able to repro within a minute or two by just scrolling continuously.

App Version: 1.7.0 from Play Store
Device: Pixel 8 running Android 15

adb logcat below shows that DBManager.setTalkProgress [calls db.close()](https://github.com/dharmaseed/dharmaseed-android/blob/master/app/src/main/java/org/dharmaseed/android/DBManager.java#L380) at the same time a ListView is trying to render using the same SQLiteConnectionPool? I noticed it's the only db.close() I could find in the project too, apart from upgrade/migration functions.

```
03-10 23:02:51.965 17310 17310 E AndroidRuntime: FATAL EXCEPTION: main
03-10 23:02:51.965 17310 17310 E AndroidRuntime: Process: org.dharmaseed.android, PID: 17310
03-10 23:02:51.965 17310 17310 E AndroidRuntime: java.lang.IllegalStateException: Cannot perform this operation because the connection pool has been closed.
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteConnectionPool.throwIfClosedLocked(SQLiteConnectionPool.java:1108)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteConnectionPool.waitForConnection(SQLiteConnectionPool.java:721)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteConnectionPool.acquireConnection(SQLiteConnectionPool.java:404)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteSession.acquireConnection(SQLiteSession.java:931)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteSession.executeForCursorWindow(SQLiteSession.java:871)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:62)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:153)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteCursor.onMove(SQLiteCursor.java:123)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:269)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at androidx.cursoradapter.widget.CursorAdapter.getItemId(CursorAdapter.java:242)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.widget.AbsListView$RecycleBin.retrieveFromScrap(AbsListView.java:7657)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.widget.AbsListView$RecycleBin.getScrapView(AbsListView.java:7368)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.widget.AbsListView.obtainView(AbsListView.java:2474)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.widget.ListView.makeAndAddView(ListView.java:2065)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.widget.ListView.fillDown(ListView.java:791)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.widget.ListView.fillGap(ListView.java:754)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.widget.AbsListView.trackMotionScroll(AbsListView.java:5617)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.widget.ListView.trackMotionScroll(ListView.java:1982)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.widget.AbsListView$FlingRunnable.run(AbsListView.java:5155)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1415)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1424)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:1024)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:949)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1398)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:991)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:232)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:317)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8787)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:871)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: Caused by: java.lang.Exception: SQLiteConnectionPool.close()
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteConnectionPool.dispose(SQLiteConnectionPool.java:272)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteConnectionPool.close(SQLiteConnectionPool.java:252)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteDatabase.dispose(SQLiteDatabase.java:559)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteDatabase.onAllReferencesReleased(SQLiteDatabase.java:536)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteClosable.releaseReference(SQLiteClosable.java:92)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at android.database.sqlite.SQLiteClosable.close(SQLiteClosable.java:124)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at org.dharmaseed.android.DBManager.setTalkProgress(DBManager.java:380)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at org.dharmaseed.android.PlaybackService.updateTalkProgress(PlaybackService.java:104)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at org.dharmaseed.android.PlaybackService.access$100(PlaybackService.java:41)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	at org.dharmaseed.android.PlaybackService$1.run(PlaybackService.java:88)
03-10 23:02:51.965 17310 17310 E AndroidRuntime: 	... 8 more
```

--- a/app/src/main/java/org/dharmaseed/android/DBManager.java
+++ b/app/src/main/java/org/dharmaseed/android/DBManager.java
@@ -377,7 +377,6 @@ public class DBManager extends AbstractDBManager {
             Log.i(LOG_TAG, "Updated talk history item for talk "+id+"("+progress+" min)");
         }
         cursor.close();
-        db.close();
     }
 
     public boolean addDownload(Talk talk) {

